### ISSUE TITLE
IDE0057: Invalid code fix with string.Substring and null-conditional operator

### ISSUE THREAD

**Issue by Tragetaschen** [dotnet/roslyn#47183]
**Version Used**: 
16.8.0 Preview 2.0

**Steps to Reproduce**:
```C#
#nullable enable
public class Test
{
    public string? M(string? arg)
        => arg?.Substring(42); // IDE0057
}
```

**Expected Behavior**:
```C#
#nullable enable
public class Test
{
    public string? M(string? arg)
        => arg?[42..]; // IDE0057
}
```

**Actual Behavior**:
The code fix generates invalid code:
```C#
#nullable enable
public class Test
{
    public static string? M(string? arg)
        => arg?.Substring[42..]; // IDE0057
}
```
Note that the `.Substring` is still there.

![image](https://user-images.githubusercontent.com/1947968/91418395-5ccb2180-e852-11ea-8e6b-5db5b1979b15.png)

**Comment by Youssef1313** [comment]
> Expected Behavior:
> I think the Analyzer should not be triggered, as there is no other way to use the range operator.

I think it should be triggered, but produces `arg?[42..]`

**Comment by CyrusNajmabadi** [comment]
@Youssef1313 's desired behavior is correct.  That's what we should be generating.
